### PR TITLE
infra: add additional support for IBM Cloud

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Adds to and updates some shell automation around IBM Cloud as well as related documentation

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: scripts/infra/README.md [hunk 1]
```diff
@@ -84,4 +84,35 @@ scripts/infra/cloud-instance.sh ec2 stop
 scripts/infra/cloud-instance.sh ec2 start
 # When you are done, you can delete the instance
 scripts/infra/cloud-instance.sh ec2 terminate
-```
\ No newline at end of file
+```
+
+### IBM Cloud Example
```
Comment (danmcp) [inline scripts/infra/README.md:89]: This is going to be troublesome to maintain given it's mostly a dup of the above example.  Can we just point out any differences (other than ibm in the command) from the above example?

**Review by danmcp — CHANGES_REQUESTED** [review]

File: scripts/infra/README.md (hunk 1 above)
Comment (nathan-weinberg) [reply scripts/infra/README.md:89]: Yup, can do :+1:

File: scripts/infra/cloud-instance.sh [hunk 2]
```diff
@@ -199,7 +199,7 @@ ibm__stop() {
 
 ibm__start() {
     handle_help_and_instance_name_opts "$@"
-    ibmcloud is instance-start -f "$INSTANCE_NAME"
+    ibmcloud is instance-start "$INSTANCE_NAME"
```
Comment (nathan-weinberg) [inline scripts/infra/cloud-instance.sh:202]: @danmcp just as a note I removed this as the `-f` flag isn't present for `instance-start`

Comment (danmcp) [reply scripts/infra/cloud-instance.sh:202]: Thanks!

File: scripts/infra/cloud-instance.sh [hunk 3]
```diff
@@ -263,6 +262,16 @@ ibm__ssh() {
...
 }
 
+ibm__scp() {
+    handle_help_and_instance_name_opts "$@"
+    shift $((OPTIND-1))
+
+    local user_name
+    user_name=$(instance_user_name "ibm")
+    ibm_calculate_instance_public_dns
+    scp -o "StrictHostKeyChecking no" -i "$IBM_KEY_LOCATION" "$@" "$user_name"@"$PUBLIC_DNS":
```
Comment (danmcp) [inline scripts/infra/cloud-instance.sh:272]: Why is this logic different?  Sure there is a good reason but I am not following from the syntax.

Comment (nathan-weinberg) [reply scripts/infra/cloud-instance.sh:272]: Hmmm I don't follow - the logic matches what's in the AWS function, just using the respective IBM variables/helper funcs

Comment (danmcp) [reply scripts/infra/cloud-instance.sh:272]: Sorry, I was looking at the wrong line.

Comment (nathan-weinberg) [reply scripts/infra/cloud-instance.sh:272]: No worries :+1:

**Review by danmcp — APPROVED** [review]

**Review by leseb — APPROVED** [review]

**Pull request merged**